core/blockchain_arbitrum.go

```diff
@@ -19,14 +19,41 @@ package core
 
 import (
 	"fmt"
+	"sync"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
 	"github.com/ethereum/go-ethereum/core/state"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
+func (bc *BlockChain) FlushTrieDB(advanceBlockChainMutex *sync.Mutex, capLimit common.StorageSize) error {
+	if bc.triedb.Scheme() == rawdb.PathScheme {
+		return nil
+	}
+
+	advanceBlockChainMutex.Lock()
+	defer advanceBlockChainMutex.Unlock()
+
+	err := bc.triedb.Commit(bc.CurrentBlock().Root, true)
```

Comment:
we don't want to commit currentBlock's root, but the oldest block we have in memory (which is the first entry in triegc)